pkg/liqoctl/move/restic.go

```diff
@@ -0,0 +1,218 @@
+						Name: "restic-registry-data",
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{
+							corev1.ReadWriteOnce,
+						},
+						Resources: corev1.ResourceRequirements{
+							Requests: corev1.ResourceList{
+								corev1.ResourceStorage: targetPvc.Spec.Resources.Requests[corev1.ResourceStorage],
+							},
+						},
+					},
+				},
+			},
+		}
+		return nil
+	})
+	if err != nil {
+		s.Fail("Failed to ensure restic repository: ", err)
+		return err
+	}
+
+	svc := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resticRegistry,
+			Namespace: liqoStorageNamespace,
+		},
+	}
+	_, err = controllerutil.CreateOrUpdate(ctx, cl, &svc, func() error {
+		svc.Spec = corev1.ServiceSpec{
+			Selector: map[string]string{
+				"app": resticRegistry,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Port:       8000,
+					TargetPort: intstr.FromInt(8000),
+					Protocol:   corev1.ProtocolTCP,
+				},
+			},
+		}
+		return nil
+	})
+	if err != nil {
+		s.Fail("Failed to ensure restic repository: ", err)
+		return err
+	}
+	s.Success("Ensured restic repository")
+	return nil
+}
+
+func deleteResticRepository(ctx context.Context, cl client.Client, printer *common.Printer) error {
+	s, err := printer.Spinner.Start("Deleting restic repository")
+	utilruntime.Must(err)
+
+	if err := cl.Delete(ctx, &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resticRegistry,
+			Namespace: liqoStorageNamespace,
+		},
+	}); client.IgnoreNotFound(err) != nil {
+		s.Fail("Failed to delete restic repository: ", err)
+		return err
+	}
+
+	if err := cl.Delete(ctx, &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      resticRegistry,
+			Namespace: liqoStorageNamespace,
+		},
+	}); client.IgnoreNotFound(err) != nil {
+		s.Fail("Failed to delete restic repository: ", err)
+		return err
+	}
+
+	err = client.IgnoreNotFound(cl.Delete(ctx, &corev1.PersistentVolumeClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "restic-registry-data-restic-registry-0",
+			Namespace: liqoStorageNamespace,
+		},
+	}))
+	if err != nil {
+		s.Fail("Failed to delete restic repository: ", err)
+	}
+
+	s.Success("Deleted restic repository")
+	return nil
+}
+
+func waitForResticRepository(ctx context.Context, cl client.Client, printer *common.Printer) error {
+	s, err := printer.Spinner.Start("Waiting for restic repository to be up and running")
+	utilruntime.Must(err)
+
+	ctx, cancel := context.WithTimeout(ctx, time.Minute*5)
+	defer cancel()
+	ticker := time.NewTicker(time.Second * 5)
+	defer ticker.Stop()
+
+	var statefulSet appsv1.StatefulSet
+	for {
+		select {
+		case <-ctx.Done():
+			s.Fail("Failed to wait for restic repository to be up and running: ", ctx.Err())
+			return ctx.Err()
+		case <-ticker.C:
+			if err := cl.Get(ctx, client.ObjectKey{
+				Name:      resticRegistry,
+				Namespace: liqoStorageNamespace,
+			}, &statefulSet); err != nil {
```

Comment:
We should not ignore the `error` here. By doing so in case of errors that are not temporary, the user will always get `timeout` error.